libraries/state_history/abi.cpp

```diff
@@ -147,6 +147,34 @@ extern const char* const state_history_plugin_abi = R"({
                 { "name": "return_value", "type": "bytes"}
             ]
         },
+        {
+            "name": "none", "fields": [
+                { "name": "prunable_digest", "type": "checksum256" }
+            ]
+        },
+        {
+            "name": "partial", "fields": [
+                { "name": "signatures", "type": "signature[]" },
+                { "name": "context_free_segments", "type": "segment_type[]" }
+            ]
+        },
+        {
+            "name": "full", "fields": [
+                { "name": "signatures", "type": "signature[]" },
+                { "name": "context_free_segments", "type": "bytes[]" }
+            ]
+        },
+        {
+            "name": "full_legacy", "fields": [
+                { "name": "signatures", "type": "signature[]" },
+                { "name": "packed_context_free_data", "type": "bytes" }
+            ]
+        },
```

Comment:
These are nested classes in ship_protocol.hpp. I would prefer to keep them with the class prunable_data_type prefix.

Reply:
Done